src/sparse_matrix.cpp

```diff
@@ -86,10 +94,19 @@ void CSRMatrix::set(unsigned i, unsigned j, const RCP<const Basic> &e)
 
     unsigned k = p_[i];
     unsigned row_end = p_[i + 1];
+    unsigned mid = k;
 
-    // TODO: Use binary search
-    while (k < row_end && j_[k] < j)
-        k++;
+    while (k < row_end) {
```

Comment:
In this search if all `j_[k] < j` then `k` is set to `row_end` which was `p_[i+1]` previously. Now it is not. I think that's why the tests fail.